### PR TITLE
store,snappy,daemon: get rid of confusing Description method in snap_local.go/snap_remote.go

### PR DESCRIPTION
**Pull request by pedronis**
This gets rid of confusing Description method in snap_local.go/snap_remote.go in favor of reading that out of snap.Infos.

### CODE REVIEW

**Comment by pedronis** [comment]
retest this please

File: store/snap_remote.go [hunk 1]
```diff
@@ -86,7 +81,8 @@ func (s *RemoteSnap) Info() *snap.Info {
 		Revision:    s.Revision(),
 		Type:        s.Type(),
 		Channel:     s.Channel(),
-		Description: s.Description(),
+		Summary:     s.Pkg.Title,       // XXX: should be summary when the store provides it
+		Description: s.Pkg.Description, // XXX not quite right but ok for now
```
Comment (niemeyer) [inline store/snap_remote.go]: Shouldn't we just break down the pkg description into first line+rest?

File: store/snap_remote.go [hunk 2]
```diff
@@ -86,7 +81,8 @@ func (s *RemoteSnap) Info() *snap.Info {
 		Revision:    s.Revision(),
 		Type:        s.Type(),
 		Channel:     s.Channel(),
-		Description: s.Description(),
+		Summary:     s.Pkg.Title,       // XXX: should be summary when the store provides it
```
Comment (niemeyer) [inline store/snap_remote.go]: Should never be the title, I think?

Comment (pedronis) [inline store/snap_remote.go]: that's what it was, until the store fixes itself not sure we can do better, we actually need this I discovered in "snap find" or use some subset of description there

File: store/snap_remote.go (hunk 1 above)
Comment (pedronis) [inline store/snap_remote.go]: well the first line afaiu is the tagline (store only concept), not the summary, so not sure it's a win atm

**Comment by niemeyer** [comment]
We agreed to have Summary in snap find, a dummy (for now) Summary field in the client, and to use description as the description and an empty summary in the store. LGTM with these changes.

**Pull request merged**